python/bin/amici_import_petab.py

```diff
@@ -138,6 +140,65 @@ def get_fixed_parameters(condition_file_name, sbml_model,
     return fixed_parameters
 
 
+def constant_species_to_parameters(sbml_model):
+    """Convert constant species in the SBML model to constant parameters.
+
+    This can be used e.g. for setting up models with condition-specific
+    constant species for PEtab, since there it is not possible to specify
+    constant species in the condition table.
+
+    Arguments:
+        sbml_model: libsbml model instance
+
+    Returns:
+        species IDs that have been turned into constants
+
+    Raises:
+
+    """
+    transformables = []
+    for species in sbml_model.getListOfSpecies():
+        if not species.getConstant() and not species.getBoundaryCondition():
+            continue
+
+        if species.getHasOnlySubstanceUnits():
+            logger.warning(
+                f"Ignoring {species.getId()} which has only substance units."
+                " Conversion not yet implemented.")
+            continue
+
+        if math.isnan(species.getInitialConcentration()):
+            logger.warning(
+                f"Ignoring {species.getId()} which has no initial "
+                "concentration. Amount conversion not yet implemented.")
+            continue
+
+        transformables.append(species.getId())
+
+    # Must not remove species while iterating over getListOfSpecies()
```

Comment:
... and turn species to constant parameter?

Reply:
not, simply not remove while iterating